src/pages/Profile/Profile.tsx

```diff
@@ -191,15 +197,23 @@ export const Profile: React.FC = () => {
     if (
       currentlySelectedOrganisation &&
       currentlySelectedOrganisation.value &&
-      currentlySelectedOrganisation.value !== profile.current_org.id
+      currentlySelectedOrganisation.value !== profile.currentOrg.id
     ) {
-      dispatch(switchOrg({ id: profile.user.id, orgId: currentlySelectedOrganisation.value }));
+      const newOrgDetails = {
+        id: currentlySelectedOrganisation.value,
+        name: currentlySelectedOrganisation.label,
+        role: currentlySelectedOrganisation.role,
+      };
+      
+      localStorage.setItem("orgInStorage", JSON.stringify(newOrgDetails));
```

Comment:
This `localSorage` seems a bit "lost" here and makes it harder to change in the future since it needs to be searched in multiple places. Move to an util as a function and use the function here.